File: molecular/src/test/java/com/hartwig/actin/molecular/orange/evidence/TestEvidenceDatabaseFactory.java
package com.hartwig.actin.molecular.orange.evidence;

import com.hartwig.actin.molecular.orange.evidence.actionability.TestActionableEventMatcherFactory;
import com.hartwig.actin.molecular.orange.evidence.known.TestKnownEventResolverFactory;

import org.jetbrains.annotations.NotNull;

public final class TestEvidenceDatabaseFactory {

    private TestEvidenceDatabaseFactory() {
    }

    @NotNull
    public static EvidenceDatabase createEmptyDatabase() {
        return new EvidenceDatabase(TestKnownEventResolverFactory.createEmpty(), TestActionableEventMatcherFactory.createEmpty());
    }

    @NotNull
    public static EvidenceDatabase createProperDatabase() {
        return new EvidenceDatabase(TestKnownEventResolverFactory.createProper(), TestActionableEventMatcherFactory.createProper());
    }
}

File: molecular/src/test/java/com/hartwig/actin/molecular/orange/evidence/TestEvidenceDatabaseFactory.kt
package com.hartwig.actin.molecular.orange.evidence

import com.hartwig.actin.molecular.orange.evidence.actionability.TestActionableEventMatcherFactory
import com.hartwig.actin.molecular.orange.evidence.known.TestKnownEventResolverFactory

object TestEvidenceDatabaseFactory {
    fun createEmptyDatabase(): EvidenceDatabase {
        return EvidenceDatabase(TestKnownEventResolverFactory.createEmpty(), TestActionableEventMatcherFactory.createEmpty())
    }

    fun createProperDatabase(): EvidenceDatabase {
        return EvidenceDatabase(TestKnownEventResolverFactory.createProper(), TestActionableEventMatcherFactory.createProper())
    }
}
